In [1]:
import pandas as pd
import numpy as np

In [2]:
rest = pd.read_csv('TA_restaurants_curated.csv',encoding='utf8', index_col=0)
rest.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 125527 entries, 0 to 1666
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Name               125527 non-null  object 
 1   City               125527 non-null  object 
 2   Cuisine Style      94176 non-null   object 
 3   Ranking            115876 non-null  float64
 4   Rating             115897 non-null  float64
 5   Price Range        77672 non-null   object 
 6   Number of Reviews  108183 non-null  float64
 7   Reviews            115911 non-null  object 
 8   URL_TA             125527 non-null  object 
 9   ID_TA              125527 non-null  object 
dtypes: float64(3), object(7)
memory usage: 10.5+ MB


In [3]:
print(len(rest))
rest.head()

125527


Name       City  \
0  Martine of Martine's Table  Amsterdam   
1         De Silveren Spiegel  Amsterdam   
2                     La Rive  Amsterdam   
3                    Vinkeles  Amsterdam   
4   Librije's Zusje Amsterdam  Amsterdam   

                                       Cuisine Style  Ranking  Rating  \
0                    ['French', 'Dutch', 'European']      1.0     5.0   
1  ['Dutch', 'European', 'Vegetarian Friendly', '...      2.0     4.5   
2  ['Mediterranean', 'French', 'International', '...      3.0     4.5   
3  ['French', 'European', 'International', 'Conte...      4.0     5.0   
4  ['Dutch', 'European', 'International', 'Vegeta...      5.0     4.5   

  Price Range  Number of Reviews  \
0    $$ - $$$              136.0   
1        $$$$              812.0   
2        $$$$              567.0   
3        $$$$              564.0   
4        $$$$              316.0   

                                             Reviews  \
0  [['Just like home', 'A Warm Welcome to Wintry ...   
1  [['Great food and staff', 'just perfect'], ['0...   
2  [['Satisfaction', 'Delicious old school restau...   
3  [['True five star dinner', 'A superb evening o...   
4  [['Best meal.... EVER', 'super food experience...   

                                              URL_TA      ID_TA  
0  /Restaurant_Review-g188590-d11752080-Reviews-M...  d11752080  
1  /Restaurant_Review-g188590-d693419-Reviews-De_...    d693419  
2  /Restaurant_Review-g188590-d696959-Reviews-La_...    d696959  
3  /Restaurant_Review-g188590-d1239229-Reviews-Vi...   d1239229  
4  /Restaurant_Review-g188590-d6864170-Reviews-Li...   d6864170

In [4]:
pd.options.display.float_format = '{:.2f}'.format
rest.describe()

Ranking    Rating  Number of Reviews
count 115876.00 115897.00          108183.00
mean    3657.46      3.99             125.18
std     3706.26      0.68             310.83
min        1.00     -1.00               2.00
25%      965.00      3.50               9.00
50%     2256.00      4.00              32.00
75%     5237.00      4.50             114.00
max    16444.00      5.00           16478.00

In [5]:
rest['Number of Reviews'] = rest['Number of Reviews'].fillna(0)
rest['Number of Reviews'] = rest['Number of Reviews'].round(0).astype('int')
rest = rest.drop_duplicates('ID_TA', keep='first')
rest.Rating.replace(-1, 0, inplace=True)

In [6]:
rest.info()
rest.dtypes

<class 'pandas.core.frame.DataFrame'>
Int64Index: 125326 entries, 0 to 1666
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Name               125326 non-null  object 
 1   City               125326 non-null  object 
 2   Cuisine Style      94047 non-null   object 
 3   Ranking            115710 non-null  float64
 4   Rating             115734 non-null  float64
 5   Price Range        77574 non-null   object 
 6   Number of Reviews  125326 non-null  int32  
 7   Reviews            115745 non-null  object 
 8   URL_TA             125326 non-null  object 
 9   ID_TA              125326 non-null  object 
dtypes: float64(2), int32(1), object(7)
memory usage: 10.0+ MB


Name                  object
City                  object
Cuisine Style         object
Ranking              float64
Rating               float64
Price Range           object
Number of Reviews      int32
Reviews               object
URL_TA                object
ID_TA                 object
dtype: object

In [7]:
rest.rename(columns={'Name': 'name',
            'City': 'city',
            'Ranking': 'ranking',
            'Rating': 'rating',
            'Reviews': 'reviews',
            'Cuisine Style':'cuisine_style',
            'Price Range':'price_range',
            'Number of Reviews':'reviews_number'}, inplace=True)

In [8]:
rest.head()

name       city  \
0  Martine of Martine's Table  Amsterdam   
1         De Silveren Spiegel  Amsterdam   
2                     La Rive  Amsterdam   
3                    Vinkeles  Amsterdam   
4   Librije's Zusje Amsterdam  Amsterdam   

                                       cuisine_style  ranking  rating  \
0                    ['French', 'Dutch', 'European']     1.00    5.00   
1  ['Dutch', 'European', 'Vegetarian Friendly', '...     2.00    4.50   
2  ['Mediterranean', 'French', 'International', '...     3.00    4.50   
3  ['French', 'European', 'International', 'Conte...     4.00    5.00   
4  ['Dutch', 'European', 'International', 'Vegeta...     5.00    4.50   

  price_range  reviews_number  \
0    $$ - $$$             136   
1        $$$$             812   
2        $$$$             567   
3        $$$$             564   
4        $$$$             316   

                                             reviews  \
0  [['Just like home', 'A Warm Welcome to Wintry ...   
1  [['Great food and staff', 'just perfect'], ['0...   
2  [['Satisfaction', 'Delicious old school restau...   
3  [['True five star dinner', 'A superb evening o...   
4  [['Best meal.... EVER', 'super food experience...   

                                              URL_TA      ID_TA  
0  /Restaurant_Review-g188590-d11752080-Reviews-M...  d11752080  
1  /Restaurant_Review-g188590-d693419-Reviews-De_...    d693419  
2  /Restaurant_Review-g188590-d696959-Reviews-La_...    d696959  
3  /Restaurant_Review-g188590-d1239229-Reviews-Vi...   d1239229  
4  /Restaurant_Review-g188590-d6864170-Reviews-Li...   d6864170

In [9]:
print("Número de Ciudades en el df: ", rest.city.nunique())
print("Número de Restaurants en el df: ", rest.ID_TA.nunique())

Número de Ciudades en el df:  31
Número de Restaurants en el df:  125326


In [10]:
df = rest.copy()

### TOP 10 Ciudades con más Restaurants

In [11]:
top_cities = (df.groupby('city')['name'].count().to_frame()).sort_values(by='name',ascending = False).head(n=10)
top_cities2 = list(top_cities.index)
top_cities2

['London',
 'Paris',
 'Madrid',
 'Barcelona',
 'Berlin',
 'Milan',
 'Rome',
 'Prague',
 'Lisbon',
 'Vienna']

In [12]:
df_filtered = df[df['city'].isin(top_cities2)]
df_filtered

name       city  \
0                        Uma  Barcelona   
1                      Viana  Barcelona   
2                     Blavis  Barcelona   
3              My Restaurant  Barcelona   
4            Bodega Biarritz  Barcelona   
...                      ...        ...   
3719  restaurant you, Vienne     Vienna   
3720       Kartonage Kitchen     Vienna   
3721                    Fino     Vienna   
3722       Cafe Zeitgenossin     Vienna   
3723               El Gaucho     Vienna   

                                          cuisine_style  ranking  rating  \
0     ['International', 'Mediterranean', 'Fusion', '...     1.00    5.00   
1     ['International', 'Mediterranean', 'Spanish', ...     2.00    5.00   
2     ['Mediterranean', 'European', 'Spanish', 'Vege...     3.00    5.00   
3     ['Mediterranean', 'European', 'Spanish', 'Vege...     4.00    5.00   
4     ['Mediterranean', 'Spanish', 'Wine Bar', 'Euro...     5.00    5.00   
...                                                 ...      ...     ...   
3719                 ['French', 'Austrian', 'European']      nan     nan   
3720  ['Austrian', 'International', 'European', 'Fus...      nan     nan   
3721                 ['German', 'Austrian', 'European']      nan     nan   
3722                  ['Cafe', 'Contemporary', 'Diner']      nan     nan   
3723                                                NaN      nan     nan   

     price_range  reviews_number  \
0           $$$$             792   
1       $$ - $$$            2707   
2       $$ - $$$             643   
3       $$ - $$$             159   
4       $$ - $$$            1078   
...          ...             ...   
3719         NaN               0   
3720           $               0   
3721         NaN               0   
3722           $               0   
3723         NaN               0   

                                                reviews  \
0     [['Perfect atmosphere and location', 'Perfecti...   
1     [['Wow! Best ever!', 'Small menu-- GET A RESER...   
2     [['The whole package', 'Incredible Service and...   
3     [['Fantastic evening!!', 'Perfect meal'], ['01...   
4     [['Deserves its reputation!', 'THESE ARE THE T...   
...                                                 ...   
3719                                                NaN   
3720                                                NaN   
3721                                                NaN   
3722                                                NaN   
3723                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187497-d8003030-Reviews-Um...   d8003030  
1     /Restaurant_Review-g187497-d8531409-Reviews-Vi...   d8531409  
2     /Restaurant_Review-g187497-d3353672-Reviews-Bl...   d3353672  
3     /Restaurant_Review-g187497-d12643500-Reviews-M...  d12643500  
4     /Restaurant_Review-g187497-d11906544-Reviews-B...  d11906544  
...                                                 ...        ...  
3719  /Restaurant_Review-g190454-d13331692-Reviews-R...  d13331692  
3720  /Restaurant_Review-g190454-d13337747-Reviews-K...  d13337747  
3721  /Restaurant_Review-g190454-d13341715-Reviews-F...  d13341715  
3722  /Restaurant_Review-g190454-d13343747-Reviews-C...  d13343747  
3723  /Restaurant_Review-g190454-d13359780-Reviews-E...  d13359780  

[83149 rows x 10 columns]

In [13]:
# Función MAP para evaluar el rango de precios en categorías.

df_filtered['price_range'] = df_filtered['price_range'].fillna('NA')
price_ranges = {'$': 'Barato', '$$ - $$$': 'Medio', '$$$$': 'Costoso', 'NA': 'SinInformacion'}
df_filtered['price_range'] = df_filtered.price_range.map(price_ranges)
df_filtered

<ipython-input-13-3e5327cdd1b0>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['price_range'] = df_filtered['price_range'].fillna('NA')
<ipython-input-13-3e5327cdd1b0>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['price_range'] = df_filtered.price_range.map(price_ranges)


name       city  \
0                        Uma  Barcelona   
1                      Viana  Barcelona   
2                     Blavis  Barcelona   
3              My Restaurant  Barcelona   
4            Bodega Biarritz  Barcelona   
...                      ...        ...   
3719  restaurant you, Vienne     Vienna   
3720       Kartonage Kitchen     Vienna   
3721                    Fino     Vienna   
3722       Cafe Zeitgenossin     Vienna   
3723               El Gaucho     Vienna   

                                          cuisine_style  ranking  rating  \
0     ['International', 'Mediterranean', 'Fusion', '...     1.00    5.00   
1     ['International', 'Mediterranean', 'Spanish', ...     2.00    5.00   
2     ['Mediterranean', 'European', 'Spanish', 'Vege...     3.00    5.00   
3     ['Mediterranean', 'European', 'Spanish', 'Vege...     4.00    5.00   
4     ['Mediterranean', 'Spanish', 'Wine Bar', 'Euro...     5.00    5.00   
...                                                 ...      ...     ...   
3719                 ['French', 'Austrian', 'European']      nan     nan   
3720  ['Austrian', 'International', 'European', 'Fus...      nan     nan   
3721                 ['German', 'Austrian', 'European']      nan     nan   
3722                  ['Cafe', 'Contemporary', 'Diner']      nan     nan   
3723                                                NaN      nan     nan   

         price_range  reviews_number  \
0            Costoso             792   
1              Medio            2707   
2              Medio             643   
3              Medio             159   
4              Medio            1078   
...              ...             ...   
3719  SinInformacion               0   
3720          Barato               0   
3721  SinInformacion               0   
3722          Barato               0   
3723  SinInformacion               0   

                                                reviews  \
0     [['Perfect atmosphere and location', 'Perfecti...   
1     [['Wow! Best ever!', 'Small menu-- GET A RESER...   
2     [['The whole package', 'Incredible Service and...   
3     [['Fantastic evening!!', 'Perfect meal'], ['01...   
4     [['Deserves its reputation!', 'THESE ARE THE T...   
...                                                 ...   
3719                                                NaN   
3720                                                NaN   
3721                                                NaN   
3722                                                NaN   
3723                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187497-d8003030-Reviews-Um...   d8003030  
1     /Restaurant_Review-g187497-d8531409-Reviews-Vi...   d8531409  
2     /Restaurant_Review-g187497-d3353672-Reviews-Bl...   d3353672  
3     /Restaurant_Review-g187497-d12643500-Reviews-M...  d12643500  
4     /Restaurant_Review-g187497-d11906544-Reviews-B...  d11906544  
...                                                 ...        ...  
3719  /Restaurant_Review-g190454-d13331692-Reviews-R...  d13331692  
3720  /Restaurant_Review-g190454-d13337747-Reviews-K...  d13337747  
3721  /Restaurant_Review-g190454-d13341715-Reviews-F...  d13341715  
3722  /Restaurant_Review-g190454-d13343747-Reviews-C...  d13343747  
3723  /Restaurant_Review-g190454-d13359780-Reviews-E...  d13359780  

[83149 rows x 10 columns]

In [41]:
rango_precio = df_filtered.groupby(['city','price_range'])[['name']].count()
rating_mean = df_filtered.groupby(['city','price_range'])[['rating']].mean()

In [42]:
rango_precio

name
city      price_range         
Barcelona Barato          1464
          Costoso          263
          Medio           3680
          SinInformacion  3018
Berlin    Barato          1153
          Costoso          134
          Medio           2366
          SinInformacion  3425
Lisbon    Barato           766
          Costoso          109
          Medio           1638
          SinInformacion  1473
London    Barato          2985
          Costoso          648
          Medio           8528
          SinInformacion  6038
Madrid    Barato          1044
          Costoso          300
          Medio           3535
          SinInformacion  4498
Milan     Barato          1241
          Costoso          243
          Medio           2337
          SinInformacion  2859
Paris     Barato          1726
          Costoso          679
          Medio           7175
          SinInformacion  5294
Prague    Barato           871
          Costoso          101
          Medio           1702
          SinInformacion  2185
Rome      Barato          1472
          Costoso          225
          Medio           3004
          SinInformacion  1248
Vienna    Barato           392
          Costoso          108
          Medio           1627
          SinInformacion  1595

In [44]:
rating_mean

rating
city      price_range           
Barcelona Barato            4.04
          Costoso           4.19
          Medio             4.00
          SinInformacion    3.86
Berlin    Barato            4.27
          Costoso           4.31
          Medio             4.08
          SinInformacion    4.10
Lisbon    Barato            4.14
          Costoso           4.14
          Medio             4.04
          SinInformacion    4.02
London    Barato            4.03
          Costoso           4.14
          Medio             3.95
          SinInformacion    3.86
Madrid    Barato            3.84
          Costoso           4.09
          Medio             3.89
          SinInformacion    3.67
Milan     Barato            3.89
          Costoso           4.11
          Medio             3.84
          SinInformacion    3.71
Paris     Barato            4.03
          Costoso           4.17
          Medio             3.95
          SinInformacion    3.89
Prague    Barato            4.12
          Costoso           4.35
          Medio             4.02
          SinInformacion    3.93
Rome      Barato            4.31
          Costoso           4.26
          Medio             4.09
          SinInformacion    4.47
Vienna    Barato            4.16
          Costoso           4.20
          Medio             4.04
          SinInformacion    4.07

In [45]:
#el sufijo es por si hay algún nombre de columna igual. No hace nada en este caso!
rango_rating = rango_precio.join(rating_mean, rsuffix = ('_precio'))
rango_rating.rename(columns = {'name' : 'num_rest', 'rating':'rating_promedio'}, inplace = True)
rango_rating.sort_index()

num_rest  rating_promedio
city      price_range                              
Barcelona Barato              1464             4.04
          Costoso              263             4.19
          Medio               3680             4.00
          SinInformacion      3018             3.86
Berlin    Barato              1153             4.27
          Costoso              134             4.31
          Medio               2366             4.08
          SinInformacion      3425             4.10
Lisbon    Barato               766             4.14
          Costoso              109             4.14
          Medio               1638             4.04
          SinInformacion      1473             4.02
London    Barato              2985             4.03
          Costoso              648             4.14
          Medio               8528             3.95
          SinInformacion      6038             3.86
Madrid    Barato              1044             3.84
          Costoso              300             4.09
          Medio               3535             3.89
          SinInformacion      4498             3.67
Milan     Barato              1241             3.89
          Costoso              243             4.11
          Medio               2337             3.84
          SinInformacion      2859             3.71
Paris     Barato              1726             4.03
          Costoso              679             4.17
          Medio               7175             3.95
          SinInformacion      5294             3.89
Prague    Barato               871             4.12
          Costoso              101             4.35
          Medio               1702             4.02
          SinInformacion      2185             3.93
Rome      Barato              1472             4.31
          Costoso              225             4.26
          Medio               3004             4.09
          SinInformacion      1248             4.47
Vienna    Barato               392             4.16
          Costoso              108             4.20
          Medio               1627             4.04
          SinInformacion      1595             4.07

### Evaluación de los rest Costosos:
- **Paris** tiene la mayor cantidad de restaurants costosos.
- **Praga** tiene la menor catidad.
- **Praga** tiene el mejor rating promedio con 4.35
- **Madrid** tiene el promedio de rating más bajo de este grupo con 4.09

In [16]:
costosos = rango_rating.xs('Costoso',level = 'price_range').sort_values(by='num_rest',ascending= False)
costosos.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')

### Evaluación de los rest Medios:
- **London** tiene la mayor cantidad de restaurants en el rango medio.
- **Vienna** tiene la menor catidad.
- **Roma** tiene el mejor rating promedio con 4.09
- **Milan** tiene el promedio de rating más bajo de este grupo con 3.83

In [17]:
medio = rango_rating.xs('Medio',level = 'price_range').sort_values(by='num_rest',ascending= False)
medio.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')

### Evaluación de los rest Baratos:
- **Londres** tiene la mayor cantidad de restaurants baratos.
- **Vienna** tiene la menor catidad.
- **Roma** tiene el mejor rating promedio con 4.31
- **Madrid** tiene el promedio de rating más bajo de este grupo con 3.83

In [18]:
baratos = rango_rating.xs('Barato',level = 'price_range').sort_values(by='num_rest',ascending= False)
baratos.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')

### Evaluación de los rest "Sin Información":
- **Londres** tiene la mayor cantidad de restaurants sin información.
- **Roma** tiene la menor catidad.
- **Roma** tiene el mejor rating promedio con 4.45
- **Madrid** tiene el promedio de rating más bajo de este grupo con 3.67

In [19]:
sininfor = rango_rating.xs('SinInformacion',level = 'price_range').sort_values(by='num_rest',ascending= False)
sininfor.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')


In [46]:
tmp = (df_filtered.groupby(['city']).agg({'rating': ['mean','max','min']}))
tmp.sort_values(by=('rating', 'mean'),ascending=False)

rating          
            mean  max  min
city                      
Rome        4.23 5.00 3.00
Berlin      4.13 5.00 0.00
Vienna      4.07 5.00 0.00
Lisbon      4.05 5.00 1.00
Prague      4.01 5.00 0.00
Barcelona   3.97 5.00 0.00
Paris       3.95 5.00 0.00
London      3.94 5.00 0.00
Milan       3.81 5.00 0.00
Madrid      3.79 5.00 0.00

Del DF anterior, podemos concluir que Roma tiene el mejor promedio (*no es sorpresa para mi*), Berlin se posiciona en el segundo lugar y Vienna en el tercero. 
Madrid se encuentra al final de esta tabla con un 3.79 en promedio.
La desviación estandar en la evaluación promedio de estos grupos es de 0.13 puntos.

In [21]:
cocinas = df_filtered.loc[:, ['city', 'name', 'cuisine_style', 'rating', 'reviews_number', 'reviews','price_range']]
#Elimino los corchetes
cocinas.cuisine_style = cocinas.cuisine_style.str.replace('[', '')
cocinas.cuisine_style = cocinas.cuisine_style.str.replace(']', '')
cocinas.columns = ['city', 'place', 'cuisine_style', 'rating', 'reviews_number', 'reviews','price_range']
#Filtro las cocinas por aquellas que tengan su rating > a 5 y numero de reviews superior a la media
cocinas = cocinas[(cocinas['reviews_number'] >= cocinas.reviews_number.mean())&(cocinas['rating'] >= 5)]
cocinas

city                      place  \
0     Barcelona                        Uma   
1     Barcelona                      Viana   
2     Barcelona                     Blavis   
3     Barcelona              My Restaurant   
4     Barcelona            Bodega Biarritz   
...         ...                        ...   
1309       Rome                Cambiamenti   
1546       Rome            Pizza & Friends   
2665       Rome  Enoteca Vini Vizi e Virtu   
0        Vienna     Casino Restaurant Wien   
183      Vienna               Senhor Vinho   

                                          cuisine_style  rating  \
0     'International', 'Mediterranean', 'Fusion', 'E...    5.00   
1     'International', 'Mediterranean', 'Spanish', '...    5.00   
2     'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
3     'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
4     'Mediterranean', 'Spanish', 'Wine Bar', 'Europ...    5.00   
...                                                 ...     ...   
1309  'Mediterranean', 'Vegetarian Friendly', 'Vegan...    5.00   
1546  'Italian', 'Pizza', 'Mediterranean', 'Vegetari...    5.00   
2665                              'Italian', 'Wine Bar'    5.00   
0     'Austrian', 'European', 'Central European', 'I...    5.00   
183   'Mediterranean', 'European', 'Portuguese', 'Ve...    5.00   

      reviews_number                                            reviews  \
0                792  [['Perfect atmosphere and location', 'Perfecti...   
1               2707  [['Wow! Best ever!', 'Small menu-- GET A RESER...   
2                643  [['The whole package', 'Incredible Service and...   
3                159  [['Fantastic evening!!', 'Perfect meal'], ['01...   
4               1078  [['Deserves its reputation!', 'THESE ARE THE T...   
...              ...                                                ...   
1309             125  [['So perfect ! Elegant and well executed foo....   
1546             123  [['Excellent and creative pizza al taglio', 'N...   
2665             132  [['Not a restaurant, but the loveliest wine s....   
0                241  [["Fine dining at it's best", 'Xmas dinner'], ...   
183              140  [['Portugal in Vienna', 'Excellent authentic P...   

     price_range  
0        Costoso  
1          Medio  
2          Medio  
3          Medio  
4          Medio  
...          ...  
1309     Costoso  
1546      Barato  
2665       Medio  
0        Costoso  
183        Medio  

[119 rows x 7 columns]

In [22]:
cocinas['city2'] = cocinas['city']
cocinas2 = cocinas.set_index('city')
cocinas2

place  \
city                                   
Barcelona                        Uma   
Barcelona                      Viana   
Barcelona                     Blavis   
Barcelona              My Restaurant   
Barcelona            Bodega Biarritz   
...                              ...   
Rome                     Cambiamenti   
Rome                 Pizza & Friends   
Rome       Enoteca Vini Vizi e Virtu   
Vienna        Casino Restaurant Wien   
Vienna                  Senhor Vinho   

                                               cuisine_style  rating  \
city                                                                   
Barcelona  'International', 'Mediterranean', 'Fusion', 'E...    5.00   
Barcelona  'International', 'Mediterranean', 'Spanish', '...    5.00   
Barcelona  'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
Barcelona  'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
Barcelona  'Mediterranean', 'Spanish', 'Wine Bar', 'Europ...    5.00   
...                                                      ...     ...   
Rome       'Mediterranean', 'Vegetarian Friendly', 'Vegan...    5.00   
Rome       'Italian', 'Pizza', 'Mediterranean', 'Vegetari...    5.00   
Rome                                   'Italian', 'Wine Bar'    5.00   
Vienna     'Austrian', 'European', 'Central European', 'I...    5.00   
Vienna     'Mediterranean', 'European', 'Portuguese', 'Ve...    5.00   

           reviews_number                                            reviews  \
city                                                                           
Barcelona             792  [['Perfect atmosphere and location', 'Perfecti...   
Barcelona            2707  [['Wow! Best ever!', 'Small menu-- GET A RESER...   
Barcelona             643  [['The whole package', 'Incredible Service and...   
Barcelona             159  [['Fantastic evening!!', 'Perfect meal'], ['01...   
Barcelona            1078  [['Deserves its reputation!', 'THESE ARE THE T...   
...                   ...                                                ...   
Rome                  125  [['So perfect ! Elegant and well executed foo....   
Rome                  123  [['Excellent and creative pizza al taglio', 'N...   
Rome                  132  [['Not a restaurant, but the loveliest wine s....   
Vienna                241  [["Fine dining at it's best", 'Xmas dinner'], ...   
Vienna                140  [['Portugal in Vienna', 'Excellent authentic P...   

          price_range      city2  
city                              
Barcelona     Costoso  Barcelona  
Barcelona       Medio  Barcelona  
Barcelona       Medio  Barcelona  
Barcelona       Medio  Barcelona  
Barcelona       Medio  Barcelona  
...               ...        ...  
Rome          Costoso       Rome  
Rome           Barato       Rome  
Rome            Medio       Rome  
Vienna        Costoso     Vienna  
Vienna          Medio     Vienna  

[119 rows x 7 columns]

In [23]:
#Maximo numero de reviews en esta lista
reviews = cocinas.groupby(['city'])[['city2','price_range','reviews_number']].max()
reviews

city2     price_range  reviews_number
city                                                
Barcelona  Barcelona           Medio            2707
Berlin        Berlin           Medio             423
Lisbon        Lisbon           Medio             423
London        London           Medio            1730
Madrid        Madrid           Medio             352
Milan          Milan           Medio             212
Paris          Paris           Medio            2305
Prague        Prague  SinInformacion             967
Rome            Rome  SinInformacion            4867
Vienna        Vienna           Medio             241

In [47]:
tmp2 = (pd.merge(reviews, cocinas2, on='reviews_number', how='left')).sort_values(by = 'reviews_number',ascending = False)
tmp2 = tmp2.set_index(np.arange(len(tmp2.index)))
tmp2

city2_x   price_range_x  reviews_number  \
0        Rome  SinInformacion            4867   
1   Barcelona           Medio            2707   
2       Paris           Medio            2305   
3      London           Medio            1730   
4      Prague  SinInformacion             967   
5      Berlin           Medio             423   
6      Berlin           Medio             423   
7      Lisbon           Medio             423   
8      Lisbon           Medio             423   
9      Madrid           Medio             352   
10     Vienna           Medio             241   
11      Milan           Medio             212   

                                   place  \
0                          Pane e Salame   
1                                  Viana   
2                                Epicure   
3                      Bar 61 Restaurant   
4               Wine O'Clock Shop Prague   
5                           Naveena Path   
6                 Ha Tapas no Mercado!!!   
7                           Naveena Path   
8                 Ha Tapas no Mercado!!!   
9                                   Cebo   
10                Casino Restaurant Wien   
11  Shaolin State of Harmony - Raw Vegan   

                                        cuisine_style  rating  \
0     'Italian', 'Street Food', 'Vegetarian Friendly'    5.00   
1   'International', 'Mediterranean', 'Spanish', '...    5.00   
2   'French', 'European', 'Vegetarian Friendly', '...    5.00   
3   'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
4   'Italian', 'Mediterranean', 'European', 'Wine ...    5.00   
5   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...    5.00   
6   'European', 'Portuguese', 'Soups', 'Mediterran...    5.00   
7   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...    5.00   
8   'European', 'Portuguese', 'Soups', 'Mediterran...    5.00   
9   'Mediterranean', 'European', 'Spanish', 'Conte...    5.00   
10  'Austrian', 'European', 'Central European', 'I...    5.00   
11  'Healthy', 'Soups', 'Vegan Options', 'Vegetari...    5.00   

                                              reviews price_range_y    city2_y  
0   [['Very nice panini', 'The best'], ['01/07/201...        Barato       Rome  
1   [['Wow! Best ever!', 'Small menu-- GET A RESER...         Medio  Barcelona  
2   [['Very nice place', 'Speechless'], ['01/08/20...       Costoso      Paris  
3   [['A Gem of a Tapas Bar!', 'Very good'], ['01/...         Medio     London  
4   [['Best dinner in prague!', 'Lovely evening'],...         Medio     Prague  
5   [['Mindblowing!', 'Authentic Indian food'], ['...        Barato     Berlin  
6   [['Lunch', 'Great meal, fantastic value!'], ['...        Barato     Lisbon  
7   [['Mindblowing!', 'Authentic Indian food'], ['...        Barato     Berlin  
8   [['Lunch', 'Great meal, fantastic value!'], ['...        Barato     Lisbon  
9   [['Absolutely divine tastes and flavours', 'Lo...       Costoso     Madrid  
10  [["Fine dining at it's best", 'Xmas dinner'], ...       Costoso     Vienna  
11  [['THE BEST RAW FOOD EVER', 'A Taste of Harmon...         Medio      Milan

In [48]:
tmp2.rename(columns={'city2_x': 'Ciudad',
            'place': 'Restaurant Sugerido',
            'price_range_x': 'Rango de precio',
            'cuisine_style': 'Estilo de Cocina',
            'reviews_number': 'Numero de Reviews'}, inplace=True)
tmp2

Ciudad Rango de precio  Numero de Reviews  \
0        Rome  SinInformacion               4867   
1   Barcelona           Medio               2707   
2       Paris           Medio               2305   
3      London           Medio               1730   
4      Prague  SinInformacion                967   
5      Berlin           Medio                423   
6      Berlin           Medio                423   
7      Lisbon           Medio                423   
8      Lisbon           Medio                423   
9      Madrid           Medio                352   
10     Vienna           Medio                241   
11      Milan           Medio                212   

                     Restaurant Sugerido  \
0                          Pane e Salame   
1                                  Viana   
2                                Epicure   
3                      Bar 61 Restaurant   
4               Wine O'Clock Shop Prague   
5                           Naveena Path   
6                 Ha Tapas no Mercado!!!   
7                           Naveena Path   
8                 Ha Tapas no Mercado!!!   
9                                   Cebo   
10                Casino Restaurant Wien   
11  Shaolin State of Harmony - Raw Vegan   

                                     Estilo de Cocina  rating  \
0     'Italian', 'Street Food', 'Vegetarian Friendly'    5.00   
1   'International', 'Mediterranean', 'Spanish', '...    5.00   
2   'French', 'European', 'Vegetarian Friendly', '...    5.00   
3   'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
4   'Italian', 'Mediterranean', 'European', 'Wine ...    5.00   
5   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...    5.00   
6   'European', 'Portuguese', 'Soups', 'Mediterran...    5.00   
7   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...    5.00   
8   'European', 'Portuguese', 'Soups', 'Mediterran...    5.00   
9   'Mediterranean', 'European', 'Spanish', 'Conte...    5.00   
10  'Austrian', 'European', 'Central European', 'I...    5.00   
11  'Healthy', 'Soups', 'Vegan Options', 'Vegetari...    5.00   

                                              reviews price_range_y    city2_y  
0   [['Very nice panini', 'The best'], ['01/07/201...        Barato       Rome  
1   [['Wow! Best ever!', 'Small menu-- GET A RESER...         Medio  Barcelona  
2   [['Very nice place', 'Speechless'], ['01/08/20...       Costoso      Paris  
3   [['A Gem of a Tapas Bar!', 'Very good'], ['01/...         Medio     London  
4   [['Best dinner in prague!', 'Lovely evening'],...         Medio     Prague  
5   [['Mindblowing!', 'Authentic Indian food'], ['...        Barato     Berlin  
6   [['Lunch', 'Great meal, fantastic value!'], ['...        Barato     Lisbon  
7   [['Mindblowing!', 'Authentic Indian food'], ['...        Barato     Berlin  
8   [['Lunch', 'Great meal, fantastic value!'], ['...        Barato     Lisbon  
9   [['Absolutely divine tastes and flavours', 'Lo...       Costoso     Madrid  
10  [["Fine dining at it's best", 'Xmas dinner'], ...       Costoso     Vienna  
11  [['THE BEST RAW FOOD EVER', 'A Taste of Harmon...         Medio      Milan

### Dónde es seguro comer:
- Los restaurants a continuación, tienen un promedio de **5** en su rating, todos tienen más comentarios de la media de comentarios de estas ciudades, por lo que serán una **apuesta segura** a la hora de visitarles! 
- Se aprecia en su columna de de reviews un sentimiento positivo por parte de los comensales y todos estos rest se encuentran en un rango medio de precio.

# Has cogido la máxima de cada ciudad! Es mas correcto decir que la lista es de los mejores valorados de cada ciudad!

In [25]:
tmp2.loc[:,['Ciudad','Rango de precio','Numero de Reviews','Restaurant Sugerido','Estilo de Cocina','reviews']]

Ciudad Rango de precio  Numero de Reviews  \
0        Rome  SinInformacion               4867   
1   Barcelona           Medio               2707   
2       Paris           Medio               2305   
3      London           Medio               1730   
4      Prague  SinInformacion                967   
5      Berlin           Medio                423   
6      Berlin           Medio                423   
7      Lisbon           Medio                423   
8      Lisbon           Medio                423   
9      Madrid           Medio                352   
10     Vienna           Medio                241   
11      Milan           Medio                212   

                     Restaurant Sugerido  \
0                          Pane e Salame   
1                                  Viana   
2                                Epicure   
3                      Bar 61 Restaurant   
4               Wine O'Clock Shop Prague   
5                           Naveena Path   
6                 Ha Tapas no Mercado!!!   
7                           Naveena Path   
8                 Ha Tapas no Mercado!!!   
9                                   Cebo   
10                Casino Restaurant Wien   
11  Shaolin State of Harmony - Raw Vegan   

                                     Estilo de Cocina  \
0     'Italian', 'Street Food', 'Vegetarian Friendly'   
1   'International', 'Mediterranean', 'Spanish', '...   
2   'French', 'European', 'Vegetarian Friendly', '...   
3   'Mediterranean', 'European', 'Spanish', 'Veget...   
4   'Italian', 'Mediterranean', 'European', 'Wine ...   
5   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...   
6   'European', 'Portuguese', 'Soups', 'Mediterran...   
7   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...   
8   'European', 'Portuguese', 'Soups', 'Mediterran...   
9   'Mediterranean', 'European', 'Spanish', 'Conte...   
10  'Austrian', 'European', 'Central European', 'I...   
11  'Healthy', 'Soups', 'Vegan Options', 'Vegetari...   

                                              reviews  
0   [['Very nice panini', 'The best'], ['01/07/201...  
1   [['Wow! Best ever!', 'Small menu-- GET A RESER...  
2   [['Very nice place', 'Speechless'], ['01/08/20...  
3   [['A Gem of a Tapas Bar!', 'Very good'], ['01/...  
4   [['Best dinner in prague!', 'Lovely evening'],...  
5   [['Mindblowing!', 'Authentic Indian food'], ['...  
6   [['Lunch', 'Great meal, fantastic value!'], ['...  
7   [['Mindblowing!', 'Authentic Indian food'], ['...  
8   [['Lunch', 'Great meal, fantastic value!'], ['...  
9   [['Absolutely divine tastes and flavours', 'Lo...  
10  [["Fine dining at it's best", 'Xmas dinner'], ...  
11  [['THE BEST RAW FOOD EVER', 'A Taste of Harmon...

In [50]:
df_filtered

name       city  \
0                        Uma  Barcelona   
1                      Viana  Barcelona   
2                     Blavis  Barcelona   
3              My Restaurant  Barcelona   
4            Bodega Biarritz  Barcelona   
...                      ...        ...   
3719  restaurant you, Vienne     Vienna   
3720       Kartonage Kitchen     Vienna   
3721                    Fino     Vienna   
3722       Cafe Zeitgenossin     Vienna   
3723               El Gaucho     Vienna   

                                          cuisine_style  ranking  rating  \
0     ['International', 'Mediterranean', 'Fusion', '...     1.00    5.00   
1     ['International', 'Mediterranean', 'Spanish', ...     2.00    5.00   
2     ['Mediterranean', 'European', 'Spanish', 'Vege...     3.00    5.00   
3     ['Mediterranean', 'European', 'Spanish', 'Vege...     4.00    5.00   
4     ['Mediterranean', 'Spanish', 'Wine Bar', 'Euro...     5.00    5.00   
...                                                 ...      ...     ...   
3719                 ['French', 'Austrian', 'European']      nan     nan   
3720  ['Austrian', 'International', 'European', 'Fus...      nan     nan   
3721                 ['German', 'Austrian', 'European']      nan     nan   
3722                  ['Cafe', 'Contemporary', 'Diner']      nan     nan   
3723                                                NaN      nan     nan   

         price_range  reviews_number  \
0            Costoso             792   
1              Medio            2707   
2              Medio             643   
3              Medio             159   
4              Medio            1078   
...              ...             ...   
3719  SinInformacion               0   
3720          Barato               0   
3721  SinInformacion               0   
3722          Barato               0   
3723  SinInformacion               0   

                                                reviews  \
0     [['Perfect atmosphere and location', 'Perfecti...   
1     [['Wow! Best ever!', 'Small menu-- GET A RESER...   
2     [['The whole package', 'Incredible Service and...   
3     [['Fantastic evening!!', 'Perfect meal'], ['01...   
4     [['Deserves its reputation!', 'THESE ARE THE T...   
...                                                 ...   
3719                                                NaN   
3720                                                NaN   
3721                                                NaN   
3722                                                NaN   
3723                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187497-d8003030-Reviews-Um...   d8003030  
1     /Restaurant_Review-g187497-d8531409-Reviews-Vi...   d8531409  
2     /Restaurant_Review-g187497-d3353672-Reviews-Bl...   d3353672  
3     /Restaurant_Review-g187497-d12643500-Reviews-M...  d12643500  
4     /Restaurant_Review-g187497-d11906544-Reviews-B...  d11906544  
...                                                 ...        ...  
3719  /Restaurant_Review-g190454-d13331692-Reviews-R...  d13331692  
3720  /Restaurant_Review-g190454-d13337747-Reviews-K...  d13337747  
3721  /Restaurant_Review-g190454-d13341715-Reviews-F...  d13341715  
3722  /Restaurant_Review-g190454-d13343747-Reviews-C...  d13343747  
3723  /Restaurant_Review-g190454-d13359780-Reviews-E...  d13359780  

[83149 rows x 10 columns]

In [26]:
citieslist = df_filtered.groupby('city')

madrid = citieslist.get_group('Madrid')
madrid.describe()

ranking  rating  reviews_number
count  8647.00 8656.00         9377.00
mean   4421.75    3.79           84.80
std    2552.74    0.75          255.25
min       1.00    0.00            0.00
25%    2189.50    3.50            4.00
50%    4453.00    4.00           18.00
75%    6638.50    4.50           70.00
max    8810.00    5.00        11497.00

In [27]:
madrid.cuisine_style = madrid.cuisine_style.str.replace('[', '')
madrid.cuisine_style = madrid.cuisine_style.str.replace(']', '')
madrid.head(n=10)

C:\Users\iwanc\anaconda3\lib\site-packages\pandas\core\generic.py:5168: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


name    city  \
0                                     Cebo  Madrid   
1                                   DSTAGE  Madrid   
2                    Los Montes de Galicia  Madrid   
3                                    Astor  Madrid   
4  Restaurante Nuevo Horno de Santa Teresa  Madrid   
5                                  Burnout  Madrid   
6                              Salmon Guru  Madrid   
7                                     Toga  Madrid   
8                                   Maitia  Madrid   
9                             Grama Lounge  Madrid   

                                       cuisine_style  ranking  rating  \
0  'Mediterranean', 'European', 'Spanish', 'Conte...     1.00    5.00   
1  'International', 'European', 'Spanish', 'Conte...     2.00    4.50   
2  'International', 'Mediterranean', 'European', ...     3.00    4.50   
3  'International', 'Mediterranean', 'European', ...     4.00    5.00   
4                         'Mediterranean', 'Spanish'     5.00    4.50   
5  'American', 'Bar', 'International', 'Healthy',...     6.00    4.50   
6                       'Bar', 'Vegetarian Friendly'     7.00    4.50   
7  'Mediterranean', 'European', 'Spanish', 'Veget...     8.00    4.50   
8  'Spanish', 'European', 'Vegetarian Friendly', ...     9.00    4.50   
9  'Latin', 'Bar', 'Venezuelan', 'Vegetarian Frie...    10.00    4.50   

  price_range  reviews_number  \
0     Costoso             352   
1     Costoso            1724   
2       Medio            3795   
3       Medio             302   
4       Medio             357   
5       Medio             413   
6       Medio             160   
7       Medio             377   
8     Costoso             690   
9       Medio             337   

                                             reviews  \
0  [['Absolutely divine tastes and flavours', 'Lo...   
1  [['Birthday meal', 'Outstanding experience. Ve...   
2  [['Wonderful lunch', 'One of Madrid’s finest.'...   
3  [['Small place with big flavours', 'The best d...   
4  [['Be hungry, be very hungry....', 'Good good'...   
5  [['When in Spain', 'Best Burger and Service in...   
6  [['Best cocktails evaaah!', 'Great'], ['01/03/...   
7  [['Well balanced', 'Fantastic food and drink c...   
8  [['Top Notch!', 'Excellent experience!'], ['12...   
9  [['Capitán Jack lives here!', 'Great Latin Ame...   

                                              URL_TA      ID_TA  
0  /Restaurant_Review-g187514-d10428302-Reviews-C...  d10428302  
1  /Restaurant_Review-g187514-d6884911-Reviews-DS...   d6884911  
2  /Restaurant_Review-g187514-d2006562-Reviews-Lo...   d2006562  
3  /Restaurant_Review-g187514-d11896546-Reviews-A...  d11896546  
4  /Restaurant_Review-g187514-d1953725-Reviews-Re...   d1953725  
5  /Restaurant_Review-g187514-d11953286-Reviews-B...  d11953286  
6  /Restaurant_Review-g187514-d10731641-Reviews-S...  d10731641  
7  /Restaurant_Review-g187514-d11955452-Reviews-T...  d11955452  
8  /Restaurant_Review-g187514-d3948218-Reviews-Ma...   d3948218  
9  /Restaurant_Review-g187514-d12216862-Reviews-G...  d12216862

In [28]:
(madrid.groupby(['price_range'])[['name']].count()).sort_values(by='name',ascending = False)

name
price_range         
SinInformacion  4498
Medio           3535
Barato          1044
Costoso          300

# A partir de aquí no creas algo eficiente. Si tu idea es sacar todos los df ya vimos como lo podríamos hacer! 

In [38]:
madrid2 = madrid.groupby(['price_range'])
madridbarato = madrid2.get_group('Barato')
madridbarato = madridbarato.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridbarato = madridbarato.set_index(np.arange(len(madridbarato.index)))
mbaratos = list(madridbarato['name'][0:15])
madridbarato



name    city  \
0                 Smok Mok  Madrid   
1         La Mula Plateada  Madrid   
2                   Plenti  Madrid   
3              Sushiberico  Madrid   
4                 Keyaan's  Madrid   
...                    ...     ...   
1039              La Manon  Madrid   
1040  Buoni Forno Di Pizze  Madrid   
1041       Espigas & Migas  Madrid   
1042         Shanghai Bund  Madrid   
1043          La Pancheria  Madrid   

                                          cuisine_style  ranking  rating  \
0     'Asian', 'Street Food', 'Vegetarian Friendly',...   551.00    5.00   
1     'Pizza', 'Argentinean', 'South American', 'Veg...   534.00    5.00   
2     'Cafe', 'Healthy', 'European', 'Vegetarian Fri...   436.00    5.00   
3              'Sushi', 'Spanish', 'Japanese', 'Fusion'  2592.00    5.00   
4     'Caribbean', 'Latin', 'Contemporary', 'Fast Fo...   451.00    5.00   
...                                                 ...      ...     ...   
1039                                    'Mediterranean'      nan     nan   
1040     'Italian', 'Pizza', 'Fast Food', 'Street Food'      nan     nan   
1041  'Brew Pub', 'Pizza', 'Cafe', 'European', 'Pub'...      nan     nan   
1042                    'Chinese', 'Fast Food', 'Asian'      nan     nan   
1043                                         'American'      nan     nan   

     price_range  reviews_number  \
0         Barato              74   
1         Barato              65   
2         Barato              63   
3         Barato              59   
4         Barato              58   
...          ...             ...   
1039      Barato               0   
1040      Barato               0   
1041      Barato               0   
1042      Barato               0   
1043      Barato               0   

                                                reviews  \
0     [['Amazing food, amazing people!!', 'premium s...   
1     [['Fantastic empanadas and pizzas, friendly a....   
2     [['aMazInG', 'Excellent eggs for breakfast'], ...   
3     [['Delicious new take on sushi'], ['12/05/2015']]   
4     [['Great taste and service', 'Excellent empana...   
...                                                 ...   
1039                                                NaN   
1040                                                NaN   
1041                                                NaN   
1042                                                NaN   
1043                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d12215222-Reviews-S...  d12215222  
1     /Restaurant_Review-g187514-d8466797-Reviews-La...   d8466797  
2     /Restaurant_Review-g187514-d12400540-Reviews-P...  d12400540  
3     /Restaurant_Review-g187514-d7689202-Reviews-Su...   d7689202  
4     /Restaurant_Review-g187514-d11756470-Reviews-K...  d11756470  
...                                                 ...        ...  
1039  /Restaurant_Review-g187514-d13206330-Reviews-L...  d13206330  
1040  /Restaurant_Review-g187514-d13273250-Reviews-B...  d13273250  
1041  /Restaurant_Review-g187514-d13273266-Reviews-E...  d13273266  
1042  /Restaurant_Review-g187514-d13294347-Reviews-S...  d13294347  
1043  /Restaurant_Review-g187514-d13307694-Reviews-L...  d13307694  

[1044 rows x 10 columns]

In [37]:
madridmedio = madrid2.get_group('Medio')
madridmedio = madridmedio.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridmedio = madridmedio.set_index(np.arange(len(madridmedio.index)))
mmedio = list(madridmedio['name'][0:15])
madridmedio

name    city  \
0                           Astor  Madrid   
1                  Meson Molinero  Madrid   
2                      Bizikletak  Madrid   
3     Malvon Empanadas Argentinas  Madrid   
4                   Picnic Madrid  Madrid   
...                           ...     ...   
3530                        Kilig  Madrid   
3531            Restaurante Sagaz  Madrid   
3532                 Flor de Jara  Madrid   
3533     Bengol Indian Restaurant  Madrid   
3534                     Gho Hfan  Madrid   

                                          cuisine_style  ranking  rating  \
0     'International', 'Mediterranean', 'European', ...     4.00    5.00   
1                               'Steakhouse', 'Spanish'    84.00    5.00   
2     'Mediterranean', 'Fusion', 'Vegetarian Friendl...   474.00    5.00   
3     'Argentinean', 'Vegetarian Friendly', 'Vegan O...   172.00    5.00   
4     'Mediterranean', 'European', 'Spanish', 'Veget...   296.00    5.00   
...                                                 ...      ...     ...   
3530                               'French', 'European'      nan     nan   
3531        'Bar', 'International', 'Spanish', 'Fusion'      nan     nan   
3532                                          'Spanish'      nan     nan   
3533                                           'Indian'      nan     nan   
3534                      'Japanese', 'Asian', 'Fusion'      nan     nan   

     price_range  reviews_number  \
0          Medio             302   
1          Medio             201   
2          Medio             151   
3          Medio             134   
4          Medio             107   
...          ...             ...   
3530       Medio               0   
3531       Medio               0   
3532       Medio               0   
3533       Medio               0   
3534       Medio               0   

                                                reviews  \
0     [['Small place with big flavours', 'The best d...   
1     [['Delicious Lamb And Excellent Service', 'Gre...   
2     [['Absolutely wonderful.', 'Hidden gem'], ['05...   
3     [['Great place for a snack', "Best empanadas I...   
4     [['Way beyond my expectations', 'Excellent fro...   
...                                                 ...   
3530                                                NaN   
3531                                                NaN   
3532                                                NaN   
3533                                                NaN   
3534                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d11896546-Reviews-A...  d11896546  
1     /Restaurant_Review-g187514-d990935-Reviews-Mes...    d990935  
2     /Restaurant_Review-g187514-d10488043-Reviews-B...  d10488043  
3     /Restaurant_Review-g187514-d11511433-Reviews-M...  d11511433  
4     /Restaurant_Review-g187514-d7898096-Reviews-Pi...   d7898096  
...                                                 ...        ...  
3530  /Restaurant_Review-g187514-d13329396-Reviews-K...  d13329396  
3531  /Restaurant_Review-g187514-d13331727-Reviews-R...  d13331727  
3532  /Restaurant_Review-g187514-d13336594-Reviews-F...  d13336594  
3533  /Restaurant_Review-g187514-d13340838-Reviews-B...  d13340838  
3534  /Restaurant_Review-g187514-d13345610-Reviews-G...  d13345610  

[3535 rows x 10 columns]

In [36]:
madridcostos = madrid2.get_group('Costoso')
madridcostos = madridcostos.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridcostos = madridcostos.set_index(np.arange(len(madridcostos.index)))
mcostoso = list(madridcostos['name'][0:15])
madridcostos

name    city  \
0                                     Cebo  Madrid   
1       Restaurante Etimo by Begona Fraire  Madrid   
2              Restaurante Crudito by Emil  Madrid   
3                                  Arrayan  Madrid   
4                               Cantespino  Madrid   
..                                     ...     ...   
295             El Culebrin Gourmet Madrid  Madrid   
296  Cafeteria Restaurante Quijote y Dulce  Madrid   
297                        Vinoteca Tierra  Madrid   
298                    Serrin Ultramarinos  Madrid   
299                   Asador Parrilla 2j's  Madrid   

                                         cuisine_style  ranking  rating  \
0    'Mediterranean', 'European', 'Spanish', 'Conte...     1.00    5.00   
1    'Spanish', 'Vegetarian Friendly', 'Gluten Free...   138.00    5.00   
2    'Japanese', 'Spanish', 'Fusion', 'Vegetarian F...   667.00    5.00   
3                           'Mediterranean', 'Spanish'  1024.00    5.00   
4                                            'Spanish'  3527.00    5.00   
..                                                 ...      ...     ...   
295                                          'Spanish'      nan     nan   
296                                          'Seafood'      nan     nan   
297                                    'Mediterranean'      nan     nan   
298                                          'Healthy'      nan     nan   
299                                         'American'      nan     nan   

    price_range  reviews_number  \
0       Costoso             352   
1       Costoso              52   
2       Costoso              31   
3       Costoso              24   
4       Costoso              22   
..          ...             ...   
295     Costoso               0   
296     Costoso               0   
297     Costoso               0   
298     Costoso               0   
299     Costoso               0   

                                               reviews  \
0    [['Absolutely divine tastes and flavours', 'Lo...   
1    [['A meal to remember, excellent experience!',...   
2    [['Beautiful uniquely Spanish Japanese fusion'...   
3    [['Amazing dinner', 'Excellent choice!!!'], ['...   
4                                             [[], []]   
..                                                 ...   
295                                                NaN   
296                                                NaN   
297                                                NaN   
298                                                NaN   
299                                                NaN   

                                                URL_TA      ID_TA  
0    /Restaurant_Review-g187514-d10428302-Reviews-C...  d10428302  
1    /Restaurant_Review-g187514-d13138415-Reviews-R...  d13138415  
2    /Restaurant_Review-g187514-d12097619-Reviews-R...  d12097619  
3    /Restaurant_Review-g187514-d12608132-Reviews-A...  d12608132  
4    /Restaurant_Review-g187514-d1991075-Reviews-Ca...   d1991075  
..                                                 ...        ...  
295  /Restaurant_Review-g187514-d12637247-Reviews-E...  d12637247  
296  /Restaurant_Review-g187514-d12664248-Reviews-C...  d12664248  
297  /Restaurant_Review-g187514-d12791037-Reviews-V...  d12791037  
298  /Restaurant_Review-g187514-d12900158-Reviews-S...  d12900158  
299  /Restaurant_Review-g187514-d12901121-Reviews-A...  d12901121  

[300 rows x 10 columns]

In [35]:
madridsininfo = madrid2.get_group('SinInformacion')
madridsininfo = madridsininfo.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridsininfo = madridsininfo.set_index(np.arange(len(madridsininfo.index)))
msininfo = list(madridsininfo['name'][0:15])
madridsininfo

name    city  \
0          Taberna la Ilustracion  Madrid   
1                     Tabernicola  Madrid   
2                          Astral  Madrid   
3     Meson Restaurante El Molino  Madrid   
4                   Chulapas Cafe  Madrid   
...                           ...     ...   
4493                 El Del Medio  Madrid   
4494                     Fit Food  Madrid   
4495                  La Revuelta  Madrid   
4496                  Casa Carmen  Madrid   
4497         El Escondite de Baco  Madrid   

                            cuisine_style  ranking  rating     price_range  \
0                       'Cafe', 'Spanish'   521.00    5.00  SinInformacion   
1                                     NaN   851.00    5.00  SinInformacion   
2                                     NaN   939.00    5.00  SinInformacion   
3                      'Spanish', 'Grill'   632.00    5.00  SinInformacion   
4                            'Bar', 'Pub'  1829.00    5.00  SinInformacion   
...                                   ...      ...     ...             ...   
4493  'Spanish', 'Contemporary', 'Fusion'      nan     nan  SinInformacion   
4494                            'Spanish'      nan     nan  SinInformacion   
4495                            'Spanish'      nan     nan  SinInformacion   
4496           'Mediterranean', 'Spanish'      nan     nan  SinInformacion   
4497                                  NaN      nan     nan  SinInformacion   

      reviews_number                                            reviews  \
0                 69                                           [[], []]   
1                 48  [['Fabulous foods and fantastic hospitality f....   
2                 41                                           [[], []]   
3                 37                                           [[], []]   
4                 31                                           [[], []]   
...              ...                                                ...   
4493               0                                                NaN   
4494               0                                                NaN   
4495               0                                                NaN   
4496               0                                                NaN   
4497               0                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d7284083-Reviews-Ta...   d7284083  
1     /Restaurant_Review-g187514-d10063815-Reviews-T...  d10063815  
2     /Restaurant_Review-g187514-d991697-Reviews-Ast...    d991697  
3     /Restaurant_Review-g187514-d12367309-Reviews-M...  d12367309  
4     /Restaurant_Review-g187514-d4366743-Reviews-Ch...   d4366743  
...                                                 ...        ...  
4493  /Restaurant_Review-g187514-d13329973-Reviews-E...  d13329973  
4494  /Restaurant_Review-g187514-d13331427-Reviews-F...  d13331427  
4495  /Restaurant_Review-g187514-d13343539-Reviews-L...  d13343539  
4496  /Restaurant_Review-g187514-d13345511-Reviews-C...  d13345511  
4497  /Restaurant_Review-g187514-d13347983-Reviews-E...  d13347983  

[4498 rows x 10 columns]

In [33]:
print('Sugerencias para comer en madrid segun los rangos de precio: ')
print('Costosos: ',mcostoso,'\nMedios: ',mmedio,'\nBaratos: ',mbaratos,'\nSinInfo: ',msininfo)


Sugerencias para comer en madrid segun los rangos de precio: 
Costosos:  ['Cebo', 'Restaurante Etimo by Begona Fraire', 'Restaurante Crudito by Emil', 'Arrayan', 'Cantespino', 'Restaurante Clos Madrid S.L', 'Kitchenclub', 'Machete Restaurante', 'Sinatra Ya', 'El Santo Baron', 'Sabor de mar', 'Gran Clavel', 'The Loft by Salsaisarten', 'Taller Silk & Spice', 'Catering Domenico'] 
Medios:  ['Astor', 'Meson Molinero', 'Bizikletak', 'Malvon Empanadas Argentinas', 'Picnic Madrid', 'Bombardino Cafe', 'The Original Globet Croquettes', 'Enklima', 'El Rincon del Cabo', 'Mercado Jamon Iberico', 'La Cocina de Gucho', 'Brew Wild Pizza Bar', 'La Escalera De Serrano', 'Mercado Tragantua', 'Francisca'] 
Baratos:  ['Smok Mok', 'La Mula Plateada', 'Plenti', 'Sushiberico', "Keyaan's", 'Kint', 'Madriles Bar', 'Macanudo Bar Espacio Arte', 'BeeHealthyShop', 'Medri Tiramisu', 'La Creperia Rosa', 'Frutas Prohibidas Juice Smoothie Coffee', 'Wok Sun', 'La Pepa Maluca', 'Denay Coffee Bar'] 
SinInfo:  ['Taberna l

# Esto es lo que hablamos... Así podias sacar los df y nombrarlos de una manera super rápida y eficiente ;P

In [66]:
listaRangoPrecios = []
for a in madrid2:
    listaRangoPrecios.append(a[0])
listaRangoPrecios

['Barato', 'Costoso', 'Medio', 'SinInformacion']

In [67]:
for precio in listaRangoPrecios:
    globals()['Madrid' + precio] = madrid2.get_group(precio)

In [68]:
MadridBarato

name    city  \
14         CalidaDe Madrid  Madrid   
33                Gandario  Madrid   
38              El miniBAR  Madrid   
59           Tinto y Tapas  Madrid   
71            La Pizzateca  Madrid   
...                    ...     ...   
9483              La Manon  Madrid   
9502  Buoni Forno Di Pizze  Madrid   
9503       Espigas & Migas  Madrid   
9518         Shanghai Bund  Madrid   
9522          La Pancheria  Madrid   

                                          cuisine_style  ranking  rating  \
14    'Mediterranean', 'European', 'Vegetarian Frien...    15.00    4.50   
33    'Fast Food', 'Bar', 'Mediterranean', 'European...    34.00    4.50   
38    'Bar', 'Mediterranean', 'Spanish', 'Vegetarian...    39.00    4.50   
59    'Mediterranean', 'European', 'Spanish', 'Veget...    60.00    4.50   
71    'Italian', 'Pizza', 'Fast Food', 'Mediterranea...    72.00    4.50   
...                                                 ...      ...     ...   
9483                                    'Mediterranean'      nan     nan   
9502     'Italian', 'Pizza', 'Fast Food', 'Street Food'      nan     nan   
9503  'Brew Pub', 'Pizza', 'Cafe', 'European', 'Pub'...      nan     nan   
9518                    'Chinese', 'Fast Food', 'Asian'      nan     nan   
9522                                         'American'      nan     nan   

     price_range  reviews_number  \
14        Barato             220   
33        Barato             341   
38        Barato             838   
59        Barato             746   
71        Barato             729   
...          ...             ...   
9483      Barato               0   
9502      Barato               0   
9503      Barato               0   
9518      Barato               0   
9522      Barato               0   

                                                reviews  \
14    [['Great Restaurant and great service !', 'Exc...   
33    [['Wonderful food, glad we walked the distanc....   
38    [['A bar, not a restaurant', 'One of my favour...   
59    [['Good simple food', 'Tasty, simple, good val...   
71    [['Good food', 'Very tasty pizza'], ['11/27/20...   
...                                                 ...   
9483                                                NaN   
9502                                                NaN   
9503                                                NaN   
9518                                                NaN   
9522                                                NaN   

                                                 URL_TA      ID_TA  
14    /Restaurant_Review-g187514-d10824797-Reviews-C...  d10824797  
33    /Restaurant_Review-g187514-d5616788-Reviews-Ga...   d5616788  
38    /Restaurant_Review-g187514-d3688138-Reviews-El...   d3688138  
59    /Restaurant_Review-g187514-d9740129-Reviews-Ti...   d9740129  
71    /Restaurant_Review-g187514-d2086638-Reviews-La...   d2086638  
...                                                 ...        ...  
9483  /Restaurant_Review-g187514-d13206330-Reviews-L...  d13206330  
9502  /Restaurant_Review-g187514-d13273250-Reviews-B...  d13273250  
9503  /Restaurant_Review-g187514-d13273266-Reviews-E...  d13273266  
9518  /Restaurant_Review-g187514-d13294347-Reviews-S...  d13294347  
9522  /Restaurant_Review-g187514-d13307694-Reviews-L...  d13307694  

[1044 rows x 10 columns]

In [69]:
MadridCostoso

name    city  \
0                                      Cebo  Madrid   
1                                    DSTAGE  Madrid   
8                                    Maitia  Madrid   
11                         La Candela Resto  Madrid   
21                                  DiverXO  Madrid   
...                                     ...     ...   
9375             El Culebrin Gourmet Madrid  Madrid   
9383  Cafeteria Restaurante Quijote y Dulce  Madrid   
9392                        Vinoteca Tierra  Madrid   
9417                    Serrin Ultramarinos  Madrid   
9420                   Asador Parrilla 2j's  Madrid   

                                          cuisine_style  ranking  rating  \
0     'Mediterranean', 'European', 'Spanish', 'Conte...     1.00    5.00   
1     'International', 'European', 'Spanish', 'Conte...     2.00    4.50   
8     'Spanish', 'European', 'Vegetarian Friendly', ...     9.00    4.50   
11           'Spanish', 'Fusion', 'Gluten Free Options'    12.00    4.50   
21    'European', 'Spanish', 'Fusion', 'Gluten Free ...    22.00    4.50   
...                                                 ...      ...     ...   
9375                                          'Spanish'      nan     nan   
9383                                          'Seafood'      nan     nan   
9392                                    'Mediterranean'      nan     nan   
9417                                          'Healthy'      nan     nan   
9420                                         'American'      nan     nan   

     price_range  reviews_number  \
0        Costoso             352   
1        Costoso            1724   
8        Costoso             690   
11       Costoso             860   
21       Costoso            1476   
...          ...             ...   
9375     Costoso               0   
9383     Costoso               0   
9392     Costoso               0   
9417     Costoso               0   
9420     Costoso               0   

                                                reviews  \
0     [['Absolutely divine tastes and flavours', 'Lo...   
1     [['Birthday meal', 'Outstanding experience. Ve...   
8     [['Top Notch!', 'Excellent experience!'], ['12...   
11    [['Amazing', 'WOW - a delight for mouth and ey...   
21    [['Speechless', 'Electrifying and surprising']...   
...                                                 ...   
9375                                                NaN   
9383                                                NaN   
9392                                                NaN   
9417                                                NaN   
9420                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d10428302-Reviews-C...  d10428302  
1     /Restaurant_Review-g187514-d6884911-Reviews-DS...   d6884911  
8     /Restaurant_Review-g187514-d3948218-Reviews-Ma...   d3948218  
11    /Restaurant_Review-g187514-d6915926-Reviews-La...   d6915926  
21    /Restaurant_Review-g187514-d7179537-Reviews-Di...   d7179537  
...                                                 ...        ...  
9375  /Restaurant_Review-g187514-d12637247-Reviews-E...  d12637247  
9383  /Restaurant_Review-g187514-d12664248-Reviews-C...  d12664248  
9392  /Restaurant_Review-g187514-d12791037-Reviews-V...  d12791037  
9417  /Restaurant_Review-g187514-d12900158-Reviews-S...  d12900158  
9420  /Restaurant_Review-g187514-d12901121-Reviews-A...  d12901121  

[300 rows x 10 columns]

In [70]:
MadridMedio

name    city  \
2                       Los Montes de Galicia  Madrid   
3                                       Astor  Madrid   
4     Restaurante Nuevo Horno de Santa Teresa  Madrid   
5                                     Burnout  Madrid   
6                                 Salmon Guru  Madrid   
...                                       ...     ...   
9533                                    Kilig  Madrid   
9536                        Restaurante Sagaz  Madrid   
9537                             Flor de Jara  Madrid   
9538                 Bengol Indian Restaurant  Madrid   
9541                                 Gho Hfan  Madrid   

                                          cuisine_style  ranking  rating  \
2     'International', 'Mediterranean', 'European', ...     3.00    4.50   
3     'International', 'Mediterranean', 'European', ...     4.00    5.00   
4                            'Mediterranean', 'Spanish'     5.00    4.50   
5     'American', 'Bar', 'International', 'Healthy',...     6.00    4.50   
6                          'Bar', 'Vegetarian Friendly'     7.00    4.50   
...                                                 ...      ...     ...   
9533                               'French', 'European'      nan     nan   
9536        'Bar', 'International', 'Spanish', 'Fusion'      nan     nan   
9537                                          'Spanish'      nan     nan   
9538                                           'Indian'      nan     nan   
9541                      'Japanese', 'Asian', 'Fusion'      nan     nan   

     price_range  reviews_number  \
2          Medio            3795   
3          Medio             302   
4          Medio             357   
5          Medio             413   
6          Medio             160   
...          ...             ...   
9533       Medio               0   
9536       Medio               0   
9537       Medio               0   
9538       Medio               0   
9541       Medio               0   

                                                reviews  \
2     [['Wonderful lunch', 'One of Madrid’s finest.'...   
3     [['Small place with big flavours', 'The best d...   
4     [['Be hungry, be very hungry....', 'Good good'...   
5     [['When in Spain', 'Best Burger and Service in...   
6     [['Best cocktails evaaah!', 'Great'], ['01/03/...   
...                                                 ...   
9533                                                NaN   
9536                                                NaN   
9537                                                NaN   
9538                                                NaN   
9541                                                NaN   

                                                 URL_TA      ID_TA  
2     /Restaurant_Review-g187514-d2006562-Reviews-Lo...   d2006562  
3     /Restaurant_Review-g187514-d11896546-Reviews-A...  d11896546  
4     /Restaurant_Review-g187514-d1953725-Reviews-Re...   d1953725  
5     /Restaurant_Review-g187514-d11953286-Reviews-B...  d11953286  
6     /Restaurant_Review-g187514-d10731641-Reviews-S...  d10731641  
...                                                 ...        ...  
9533  /Restaurant_Review-g187514-d13329396-Reviews-K...  d13329396  
9536  /Restaurant_Review-g187514-d13331727-Reviews-R...  d13331727  
9537  /Restaurant_Review-g187514-d13336594-Reviews-F...  d13336594  
9538  /Restaurant_Review-g187514-d13340838-Reviews-B...  d13340838  
9541  /Restaurant_Review-g187514-d13345610-Reviews-G...  d13345610  

[3535 rows x 10 columns]

In [71]:
MadridSinInformacion

name    city  \
80                  Oven Gran Via  Madrid   
118                      StreetXo  Madrid   
247                  Divino Tinto  Madrid   
520        Taberna la Ilustracion  Madrid   
631   Meson Restaurante El Molino  Madrid   
...                           ...     ...   
9534                 El Del Medio  Madrid   
9535                     Fit Food  Madrid   
9539                  La Revuelta  Madrid   
9540                  Casa Carmen  Madrid   
9542         El Escondite de Baco  Madrid   

                                          cuisine_style  ranking  rating  \
80    'Italian', 'Pizza', 'Mediterranean', 'European...    81.00    4.50   
118   'International', 'Asian', 'Fusion', 'Gluten Fr...   119.00    4.50   
247                                                 NaN   248.00    4.50   
520                                   'Cafe', 'Spanish'   521.00    5.00   
631                                  'Spanish', 'Grill'   632.00    5.00   
...                                                 ...      ...     ...   
9534                'Spanish', 'Contemporary', 'Fusion'      nan     nan   
9535                                          'Spanish'      nan     nan   
9539                                          'Spanish'      nan     nan   
9540                         'Mediterranean', 'Spanish'      nan     nan   
9542                                                NaN      nan     nan   

         price_range  reviews_number  \
80    SinInformacion            1514   
118   SinInformacion            3114   
247   SinInformacion             104   
520   SinInformacion              69   
631   SinInformacion              37   
...              ...             ...   
9534  SinInformacion               0   
9535  SinInformacion               0   
9539  SinInformacion               0   
9540  SinInformacion               0   
9542  SinInformacion               0   

                                                reviews  \
80    [['Very good quality', 'Good food for a good p...   
118   [['One of the best dining experiences ever!!',...   
247   [['Friendly atmosphere and amazing food', 'Rea...   
520                                            [[], []]   
631                                            [[], []]   
...                                                 ...   
9534                                                NaN   
9535                                                NaN   
9539                                                NaN   
9540                                                NaN   
9542                                                NaN   

                                                 URL_TA      ID_TA  
80    /Restaurant_Review-g187514-d8735932-Reviews-Ov...   d8735932  
118   /Restaurant_Review-g187514-d3674644-Reviews-St...   d3674644  
247   /Restaurant_Review-g187514-d8727668-Reviews-Di...   d8727668  
520   /Restaurant_Review-g187514-d7284083-Reviews-Ta...   d7284083  
631   /Restaurant_Review-g187514-d12367309-Reviews-M...  d12367309  
...                                                 ...        ...  
9534  /Restaurant_Review-g187514-d13329973-Reviews-E...  d13329973  
9535  /Restaurant_Review-g187514-d13331427-Reviews-F...  d13331427  
9539  /Restaurant_Review-g187514-d13343539-Reviews-L...  d13343539  
9540  /Restaurant_Review-g187514-d13345511-Reviews-C...  d13345511  
9542  /Restaurant_Review-g187514-d13347983-Reviews-E...  d13347983  

[4498 rows x 10 columns]

--------------------------------------------------------------

In [40]:
tmp5 = madridcostos.append(madridmedio, ignore_index=True).append(madridbarato,ignore_index = True).append(madridsininfo,ignore_index = True)
tmp5 = tmp5.set_index(np.arange(len(tmp5.index)))
tmp5 = tmp5.dropna(subset=['cuisine_style']) 
tmp5.reset_index(drop=True,inplace=True) 
tmp5

name    city  \
0                                   Cebo  Madrid   
1     Restaurante Etimo by Begona Fraire  Madrid   
2            Restaurante Crudito by Emil  Madrid   
3                                Arrayan  Madrid   
4                             Cantespino  Madrid   
...                                  ...     ...   
6283                          El Secreto  Madrid   
6284                        El Del Medio  Madrid   
6285                            Fit Food  Madrid   
6286                         La Revuelta  Madrid   
6287                         Casa Carmen  Madrid   

                                          cuisine_style  ranking  rating  \
0     'Mediterranean', 'European', 'Spanish', 'Conte...     1.00    5.00   
1     'Spanish', 'Vegetarian Friendly', 'Gluten Free...   138.00    5.00   
2     'Japanese', 'Spanish', 'Fusion', 'Vegetarian F...   667.00    5.00   
3                            'Mediterranean', 'Spanish'  1024.00    5.00   
4                                             'Spanish'  3527.00    5.00   
...                                                 ...      ...     ...   
6283                                 'Spanish', 'Diner'      nan     nan   
6284                'Spanish', 'Contemporary', 'Fusion'      nan     nan   
6285                                          'Spanish'      nan     nan   
6286                                          'Spanish'      nan     nan   
6287                         'Mediterranean', 'Spanish'      nan     nan   

         price_range  reviews_number  \
0            Costoso             352   
1            Costoso              52   
2            Costoso              31   
3            Costoso              24   
4            Costoso              22   
...              ...             ...   
6283  SinInformacion               0   
6284  SinInformacion               0   
6285  SinInformacion               0   
6286  SinInformacion               0   
6287  SinInformacion               0   

                                                reviews  \
0     [['Absolutely divine tastes and flavours', 'Lo...   
1     [['A meal to remember, excellent experience!',...   
2     [['Beautiful uniquely Spanish Japanese fusion'...   
3     [['Amazing dinner', 'Excellent choice!!!'], ['...   
4                                              [[], []]   
...                                                 ...   
6283                                                NaN   
6284                                                NaN   
6285                                                NaN   
6286                                                NaN   
6287                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d10428302-Reviews-C...  d10428302  
1     /Restaurant_Review-g187514-d13138415-Reviews-R...  d13138415  
2     /Restaurant_Review-g187514-d12097619-Reviews-R...  d12097619  
3     /Restaurant_Review-g187514-d12608132-Reviews-A...  d12608132  
4     /Restaurant_Review-g187514-d1991075-Reviews-Ca...   d1991075  
...                                                 ...        ...  
6283  /Restaurant_Review-g187514-d13328766-Reviews-E...  d13328766  
6284  /Restaurant_Review-g187514-d13329973-Reviews-E...  d13329973  
6285  /Restaurant_Review-g187514-d13331427-Reviews-F...  d13331427  
6286  /Restaurant_Review-g187514-d13343539-Reviews-L...  d13343539  
6287  /Restaurant_Review-g187514-d13345511-Reviews-C...  d13345511  

[6288 rows x 10 columns]

# Aquí te pongo una manera más eficiente de  hacer esto desde un principio con un bucle for. 



In [60]:
tmpFacil = pd.DataFrame()
for g in madrid2:
    #print(g[1])
    #print('-----------------')
    tmpFacil= tmpFacil.append(g[1], ignore_index = True)
tmpFacil


name    city  \
0          CalidaDe Madrid  Madrid   
1                 Gandario  Madrid   
2               El miniBAR  Madrid   
3            Tinto y Tapas  Madrid   
4             La Pizzateca  Madrid   
...                    ...     ...   
9372          El Del Medio  Madrid   
9373              Fit Food  Madrid   
9374           La Revuelta  Madrid   
9375           Casa Carmen  Madrid   
9376  El Escondite de Baco  Madrid   

                                          cuisine_style  ranking  rating  \
0     'Mediterranean', 'European', 'Vegetarian Frien...    15.00    4.50   
1     'Fast Food', 'Bar', 'Mediterranean', 'European...    34.00    4.50   
2     'Bar', 'Mediterranean', 'Spanish', 'Vegetarian...    39.00    4.50   
3     'Mediterranean', 'European', 'Spanish', 'Veget...    60.00    4.50   
4     'Italian', 'Pizza', 'Fast Food', 'Mediterranea...    72.00    4.50   
...                                                 ...      ...     ...   
9372                'Spanish', 'Contemporary', 'Fusion'      nan     nan   
9373                                          'Spanish'      nan     nan   
9374                                          'Spanish'      nan     nan   
9375                         'Mediterranean', 'Spanish'      nan     nan   
9376                                                NaN      nan     nan   

         price_range  reviews_number  \
0             Barato             220   
1             Barato             341   
2             Barato             838   
3             Barato             746   
4             Barato             729   
...              ...             ...   
9372  SinInformacion               0   
9373  SinInformacion               0   
9374  SinInformacion               0   
9375  SinInformacion               0   
9376  SinInformacion               0   

                                                reviews  \
0     [['Great Restaurant and great service !', 'Exc...   
1     [['Wonderful food, glad we walked the distanc....   
2     [['A bar, not a restaurant', 'One of my favour...   
3     [['Good simple food', 'Tasty, simple, good val...   
4     [['Good food', 'Very tasty pizza'], ['11/27/20...   
...                                                 ...   
9372                                                NaN   
9373                                                NaN   
9374                                                NaN   
9375                                                NaN   
9376                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d10824797-Reviews-C...  d10824797  
1     /Restaurant_Review-g187514-d5616788-Reviews-Ga...   d5616788  
2     /Restaurant_Review-g187514-d3688138-Reviews-El...   d3688138  
3     /Restaurant_Review-g187514-d9740129-Reviews-Ti...   d9740129  
4     /Restaurant_Review-g187514-d2086638-Reviews-La...   d2086638  
...                                                 ...        ...  
9372  /Restaurant_Review-g187514-d13329973-Reviews-E...  d13329973  
9373  /Restaurant_Review-g187514-d13331427-Reviews-F...  d13331427  
9374  /Restaurant_Review-g187514-d13343539-Reviews-L...  d13343539  
9375  /Restaurant_Review-g187514-d13345511-Reviews-C...  d13345511  
9376  /Restaurant_Review-g187514-d13347983-Reviews-E...  d13347983  

[9377 rows x 10 columns]

In [63]:
tmpFacil.dropna(subset=['cuisine_style'],inplace=True)
tmpFacil.reset_index(drop = True,inplace = True)
tmpFacil

name    city  \
0     CalidaDe Madrid  Madrid   
1            Gandario  Madrid   
2          El miniBAR  Madrid   
3       Tinto y Tapas  Madrid   
4        La Pizzateca  Madrid   
...               ...     ...   
6283       El Secreto  Madrid   
6284     El Del Medio  Madrid   
6285         Fit Food  Madrid   
6286      La Revuelta  Madrid   
6287      Casa Carmen  Madrid   

                                          cuisine_style  ranking  rating  \
0     'Mediterranean', 'European', 'Vegetarian Frien...    15.00    4.50   
1     'Fast Food', 'Bar', 'Mediterranean', 'European...    34.00    4.50   
2     'Bar', 'Mediterranean', 'Spanish', 'Vegetarian...    39.00    4.50   
3     'Mediterranean', 'European', 'Spanish', 'Veget...    60.00    4.50   
4     'Italian', 'Pizza', 'Fast Food', 'Mediterranea...    72.00    4.50   
...                                                 ...      ...     ...   
6283                                 'Spanish', 'Diner'      nan     nan   
6284                'Spanish', 'Contemporary', 'Fusion'      nan     nan   
6285                                          'Spanish'      nan     nan   
6286                                          'Spanish'      nan     nan   
6287                         'Mediterranean', 'Spanish'      nan     nan   

         price_range  reviews_number  \
0             Barato             220   
1             Barato             341   
2             Barato             838   
3             Barato             746   
4             Barato             729   
...              ...             ...   
6283  SinInformacion               0   
6284  SinInformacion               0   
6285  SinInformacion               0   
6286  SinInformacion               0   
6287  SinInformacion               0   

                                                reviews  \
0     [['Great Restaurant and great service !', 'Exc...   
1     [['Wonderful food, glad we walked the distanc....   
2     [['A bar, not a restaurant', 'One of my favour...   
3     [['Good simple food', 'Tasty, simple, good val...   
4     [['Good food', 'Very tasty pizza'], ['11/27/20...   
...                                                 ...   
6283                                                NaN   
6284                                                NaN   
6285                                                NaN   
6286                                                NaN   
6287                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d10824797-Reviews-C...  d10824797  
1     /Restaurant_Review-g187514-d5616788-Reviews-Ga...   d5616788  
2     /Restaurant_Review-g187514-d3688138-Reviews-El...   d3688138  
3     /Restaurant_Review-g187514-d9740129-Reviews-Ti...   d9740129  
4     /Restaurant_Review-g187514-d2086638-Reviews-La...   d2086638  
...                                                 ...        ...  
6283  /Restaurant_Review-g187514-d13328766-Reviews-E...  d13328766  
6284  /Restaurant_Review-g187514-d13329973-Reviews-E...  d13329973  
6285  /Restaurant_Review-g187514-d13331427-Reviews-F...  d13331427  
6286  /Restaurant_Review-g187514-d13343539-Reviews-L...  d13343539  
6287  /Restaurant_Review-g187514-d13345511-Reviews-C...  d13345511  

[6288 rows x 10 columns]

## Aunque en realidad esto que has hecho al final no tiene mucho sentido... Podias extraer desde cualquiera de los df anteriores a todo esto solo las filas en las cuales la ciudad sea Madid... Pero almenos espero que veas como sacar de una serie todos los dataframes! 

Muy chulo todo, excepto lo de no resetear índices usando el reset_index y los problemillas de eficiencia lo demás está genial! Espero con inquietud el otro df!